src/main/java/com/google/javascript/gents/GentsCodeGenerator.java

```diff
@@ -54,11 +56,20 @@ protected void add(Node n, Context ctx) {
         break;
     }
 
-    // Add an empty line if the next node is further away than the next line.
+    addNewlines(n);
+  }
+
+  /** Add newlines to the generated source */
+  private void addNewlines(Node n) {
     Node nextNode = n.getNext();
     if (nextNode != null) {
       int lineSpacing = nextNode.getLineno() - n.getLineno();
-      if (lineSpacing > 1) {
+
+      // Add an empty line if:
+      //   - the next node is further away than the next line
+      //   - the next node is a class function (but the current node is not a comment)
+      if (lineSpacing > 1 ||
+          (nextNode.getToken() == Token.MEMBER_FUNCTION_DEF ^ n.getToken() == Token.EMPTY)) {
```

Comment:
Otherwise, the output will have an extra new line after the `//inline comment`

See below:

Input:
```javascript
const F = goog.defineClass(null, {
  // inline comment

  /**
   * block comment
   */
  constructor: function() {
  },
});
```

Output without check:
```typescript
class F {
  // inline comment


  /**
   * block comment
   */
  constructor() {}
}
```